### PR TITLE
FIPS support for the openstack-operators

### PR DESCRIPTION
**Pull request by fmount**
This patch adds general information and the steps required to build a fips compatible operator images.

### CODE REVIEW

File: fips.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+their algorithms properly. In the operator-k8s context, a given operator can be
+considered FIPS compliant if the underlying components are FIPS validated.
+
+
+## Build a FIPS compliant operator
+
+To be Fips compliant, there are a few requirements and recommendation that
+should be take into account:
+
+- use RHEL based images with openssl to built the binary (alpine or ubuntu images
```
Comment (abays) [inline fips.md:15]: nit: `s/use/Use/`

**Review by abays — APPROVED** [review]
/lgtm

File: fips.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+## General information
+
+FIPS 140-2 and 140-3 are designed to ensure that cryptographic tools implement
+their algorithms properly. In the operator-k8s context, a given operator can be
+considered FIPS compliant if the underlying components are FIPS validated.
+
+
+## Build a FIPS compliant operator
+
+To be Fips compliant, there are a few requirements and recommendation that
```
Comment (slawqo) [inline fips.md:12]: nit: s/Fips/FIPS

**Review by rebtoor — APPROVED** [review]

File: fips.md (hunk 1 above)
Comment (fmount) [reply fips.md]: Thank you, fixed!

File: fips.md (hunk 2 above)
Comment (fmount) [reply fips.md]: Thank you, fixed now!

File: fips.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+# FIPS Support for the openstack operators
+
+## General information
+
+FIPS 140-2 and 140-3 are designed to ensure that cryptographic tools implement
+their algorithms properly. In the operator-k8s context, a given operator can be
+considered FIPS compliant if the underlying components are FIPS validated.
```
[the remark this answers is not in the document]
Comment (fmount) [reply fips.md:7]: +1 to expand it, I didn't want to dig more into this section apart what FIPS means for a "component" that should be validated. Feel free to review/expand in a follow up patch.

File: fips.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+- Go containers rely on OpenSSL to detect whether the system is in FIPS mode
+
+
+### Steps to build a FIPS compatible operator
+
+In the `Dockerfile`, replace both the `GOLANG_BUILDER` and the `OPERATOR_BASE`
+image to point to the `go-toolset` and the `ubi-minimal` image:
+
+```bash
+ARG GOLANG_BUILDER=registry.access.redhat.com/ubi9/go-toolset:1.19
```
[the remark this answers is not in the document]
Comment (fmount) [reply fips.md:28]: this is not an internal registry [1] and does not require authentication.
More info in this article that can be useful [2]

[1] https://catalog.redhat.com/software/containers/ubi9/go-toolset/61e5c00b4ec9945c18787690
[2] https://developers.redhat.com/blog/2019/09/05/red-hat-openshift-4-on-your-laptop-introducing-red-hat-codeready-containers

**Pull request merged**